src/lib/core/common-behaviors/color.ts

```diff
@@ -0,0 +1,47 @@
+import {Constructor} from './constructor';
+import {ElementRef, Renderer2} from '@angular/core';
+
+/** List of possible color values that can be set. */
+const AVAILABLE_COLOR_VALUES = ['primary', 'accent', 'warn'];
+
+/** @docs-private */
+export interface IsColorable {
+  color: string;
+}
+
+/** @docs-private */
+export interface ColorableBase {
+  _renderer: Renderer2;
+  _elementRef: ElementRef;
+}
+
+/** Mixin to augment a directive with a `color` property. */
+export function mixinColor<T extends Constructor<ColorableBase>>(base: T, allowNoColor = false)
+    : Constructor<IsColorable> & T {
+  return class extends base {
+    private _color: string = null;
+
+    constructor(...args: any[]) { super(...args); }
+
+    get color(): string { return this._color; }
+    set color(value: string) {
+      if (AVAILABLE_COLOR_VALUES.indexOf(value) !== -1 || (allowNoColor && !value)) {
+        this._setColorClass(this._color, false);
+        this._setColorClass(value, true);
+        this._color = value;
+      }
+    }
+
+    /** Method that changes the color classes on the host element. */
+    private _setColorClass(colorName: string, isAdd: boolean) {
```

Comment:
The main reason I don't like it is the `isAdd` argument:
https://martinfowler.com/bliki/FlagArgument.html

So my thinking is "How do we avoid the boolean argument? Well, it would just be two different methods. Oh, well we already have that on renderer, so just use that."